### ISSUE TITLE
ImageDimensionBear

### ISSUE THREAD

**Issue by jayvdb** [coala/coala-bears#1260]
c.f https://github.com/coala/coala-bears/issues/1259

https://github.com/fossasia/gci16.fossasia.org/issues/239 was a feature request to add an image size linter.
FOSSASIA contributor @Abhi2424shek  created the tool in https://github.com/fossasia/gci16.fossasia.org/pull/467 .

History of it is split between https://github.com/fossasia/gci16.fossasia.org/commits/3ba48810d46416f32267c3f99f4d9498ac1399da/scripts/ImgChecker/ImgChecker.rb (with some enhancements by @ankitrgadiya) and https://github.com/fossasia/gci16.fossasia.org/commits/gh-pages/scripts/image_checker.rb after @robbyoconnor renamed it and radically improved it.

The current script is at:
https://github.com/fossasia/gci16.fossasia.org/blob/gh-pages/scripts/image_checker.rb

This issue is to create a similar coala bear.

Ideally the authors of that tool come together and publish their code in a separate ruby gem, so it can be used by coala.

**Comment by robbyoconnor** [comment]
@jayvdb -- that script would need to be radically refactored to handle non-square dimensions...I have a few ideas on how to do this. I'll work on it later today.

**Comment by robbyoconnor** [comment]
[The current script](https://github.com/fossasia/gci16.fossasia.org/tree/gh-pages/scripts/image_checker.rb) is highly specialized for FOSSASIA's GCI16 site -- I could probably make this more generic and package it up as a gem. I do not think this should be a GCI task -- students need to work on harder things right now...I'll help out with this one.

**Comment by robbyoconnor** [comment]
What would be REALLY cool with this bear existing is that it can actually resize the images for you!

**Comment by Abhi2424shek** [comment]
Is there a task for it @jayvdb

**Comment by robbyoconnor** [comment]
@Abhi2424shek -- if you wanna work on this -- I'll help you out -- the main reason I did what I did and rewrote it was I saw a lot of repeated code. I'd like that to remain. 

As far as stuff I did to it - a lot of my work was simply DRY'ing up the code and bringing it in-line with the ruby style guide.

**Comment by robbyoconnor** [comment]
As far as license, it'd have to be GPL. It's a derivative work.

**Comment by robbyoconnor** [comment]
@jayvdb -- that work?

**Comment by Abhi2424shek** [comment]
@robbyoconnor @jayvdb can I sir, but I think it's more good to create a repository and where we three of us (myself, @ankitrgadiya and @robbyoconnor) could code together and create the ruby gem there for this issue.

**Comment by Abhi2424shek** [comment]
@jayvdb can you share the link to the gci task for us pls..

**Comment by jayvdb** [comment]
The task for the ruby gem was created a few days ago, and is still unpublished, but here is the link https://codein.withgoogle.com/tasks/5746072943067136/

The task for the coala bear which uses the ruby gem has just been created so it is also unpublished, but again here is the link https://codein.withgoogle.com/tasks/6269128656551936/

**Comment by Abhi2424shek** [comment]
@jayvdb thanks we will just accept it when the task comes alive!

**Comment by Abhi2424shek** [comment]
@jayvdb the gems published;  link to repo: https://github/Abhi2424shek/img_checker

Install gem by: gem install img_checker

Check the gem: https://rubygems.org/gems/img_checker

**Comment by ananyo2012** [comment]
@jayvdb Just being curious how are rubygems useful in the making of a bear? Also will this bear for serving its purpose for linting in any language or will it be ruby specific? Pardon me if I am wrong saying anything as I am not acquainted with coala-bears. I just seem to have a rough idea seeing the repository structure.

**Comment by robbyoconnor** [comment]
In the same way that rubocop is used. It provides a means to get the linter.

**Comment by robbyoconnor** [comment]
@ananyo2012, In the same way that rubocop is used. It provides a means to get the linter. It allows others outside of coala to use it as well.

**Comment by ananyo2012** [comment]
@robbyoconnor Yes use outside coala as a gem is fine. I was asking with relation to use in coala-bears. rubocop works as a ruby linter only. So is this something as ruby specific or can we use it in generalized way?

**Comment by robbyoconnor** [comment]
@ananyo2012 the linter happens to be a Ruby script. The Image Linter can be used outside of Ruby -- it literally checks the image dimensions (for now). It could be extended to check image size as well.

**Comment by nemaniarjun** [comment]
@ananyo2012 coala has a LinterBear subclass. Which can run any executable and parse its output.
I assume that this is what @robbyoconnor will use for this bear.

**Comment by robbyoconnor** [comment]
@nemaniarjun is correct. That's the reason it's packaged as a gem.

**Comment by jayvdb** [comment]
fwiw, a working prototype is at https://github.com/coala/coala-bears/pull/1288

**Comment by jayvdb** [comment]
This bear could have been used to prevent https://github.com/coala/coala/pull/4078 from being merged , and check https://github.com/coala/coala/pull/4081 .
